libraries/c_sdk/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -175,6 +175,27 @@ bool _IotMqtt_ValidateConnect( const IotMqttConnectInfo_t * pConnectInfo )
         EMPTY_ELSE_MARKER;
     }
 
+    /* Check that keep alive is not too short. */
+    if( pConnectInfo->keepAliveSeconds != 0 )
+    {
+        if( ( pConnectInfo->keepAliveSeconds * 1000 ) <= IOT_MQTT_RESPONSE_WAIT_MS )
```

Comment:
Please write a comment here why we are putting this restriction. Also if keepAliveSeconds is zero, for awsIotMqttMode, it means it is AWS_IOT_MQTT_SERVER_MAX_KEEPALIVE, so we can check against that value if awsIotMqttMode is enabled.  Otherwise keep-alive is disabled with 0 value and non-aws mode.

Reply:
Changed to check appropriate interval with awsIotMqttMode enabled